draft-protected-headers.md

```diff
@@ -28,58 +32,192 @@ author:
+This document describes a common strategy to extend the cryptographic protections provided by PGP/MIME etc. to also protect message headers. In particular, how to encrypt the Subject line.
 
 --- middle
 
 Introduction
 ============
 
-FIXME
+The standards for cryptographic e-mail, PGP/MIME and S/MIME ({{RFC3156}} and {{RFC8551}}), can provide both integrity and confidentiality to the body of a MIME e-mail message. However, PGP/MIME alone provides no protection to message headers and the mechanism specified for header protection within S/MIME has not seen widespread adoption.
+
+This document defines a scheme, "Protected Headers for Cryptographic E-mail", which has been adopted by multiple existing e-mail clients in order to extend the cryptographic protections provided by PGP/MIME to also protect the {{RFC2822}} message headers.
+
+In particular, we describe how to encrypt the Subject line and how to preserve backwards compatibility so an encrypted subject remains available to recipients using software that does not implement support for the Protected Headers scheme.
+
+We also discuss some of the compatibility constraints and usability concerns which motivated the design of the scheme, as well as limitations.
+
+The authors believe the technique is broadly applicable would also apply to other MIME-compatible cryptographic e-mail systems, including S/MIME.  Furthermore, this technique has already proven itself as a useful building block for other improvements to cryptographic e-mail, such as the Autocrypt Level 1.1 "Gossip" mechanism.
+
 
 Requirements Language
 ---------------------
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14 {{RFC2119}} {{RFC8174}} when, and only when, they appear in all capitals, as shown here.
 
-Cryptographic Payload
-=====================
 
-Message Composition (Header Copying)
-====================================
+Terminology
+-----------
+
+For the purposes of this document, we define the following concepts:
+
+   * *MUA* is short for Mail User Agent; an e-mail client.
+   * *Protection* of message data refers to cryptographic encryption and/or signatures, providing authenticity, confidentiality or both.
+   * *Cryptographic Envelope* is all MIME structure directly dictated by the cryptographic e-mail system in use.
+   * *Cryptographic Payload* is all message data protected by the Cryptographic Envelope.
+   * *Original Headers* are the RFC2822 message headers as known to the sending MUA at the time of message composition.
+   * *Protected Headers* are any headers protected by the scheme described in this document.
+   * *Exposed Headers* are any headers outside the Cryptographic Payload (protected or not).
+   * *Obscured Headers* are any headers which have been modified or removed from the set of Exposed Headers.
```

Comment:
I prefer obscured, because that communicates the intent of WHY we are modifying them. We're not modifying them just-because, we are doing so to reduce exposure of sensitive data. Happy to hear other points of view though.

Reply:
Agree with your point